webapp/src/components/form/FormCreator.js

```diff
@@ -0,0 +1,310 @@
+        event: res.event,
+        error: res.error
+      })
+    });
+  }, []);
+
+  const handleCheckChanged = (e) => {
+    const val = e.target.checked;
+    setTimeout(() => {
+      setNominate(val);
+    }, 1);
+  }
+
+  const handleLanguageChange = (e) => {
+    setLanguage(e);
+  }
+
+  const handleSection = (input) => {
+    setSections(input);
+  }
+
+  const handleDrag = (e) => {
+    if(parentDropable) {
+      drag(e, setDragId);
+    }
+  }
+
+  const handleDragOver = (e) => {
+    e.preventDefault();
+  };
+
+  const handleDrop = (e) => {
+    if (parentDropable) {
+      drop({
+        event: e,
+        elements: sections,
+        dragId,
+        setState: setSections,
+        setAnimation: setApplytransition
+      });
+    }
+  }
+
+  const options = () => {
+    return lang.map(l => option({
+      value: l.code,
+      label: l.description,
+      t
+    }));
+  }
+
+  let isSaveDisabled = false;
+  sections.forEach(s => {
+    for(let key of Object.keys(s.name)) {
+      if (!s.name[key]) {
+        isSaveDisabled = true;
+      }
+    }
+    s.questions.forEach(q => {
+      for(let key of Object.keys(q.headline)) {
+        if (!q.headline[key]) {
+          isSaveDisabled = true;
+        }
+      }
+      if (!q.type) {
+        isSaveDisabled = true;
+      }
+    })
+  })
+
+  
+
+  const {loading, event: evnt, error } = events;
+
+  if (loading) {
+    return <Loading />
+  }
+  if (error) {
+    return (
+      <div className='alert alert-danger alert-container'>
+        {error}
+      </div>
+    )
+  }
+  return (
+    <>
+      <Prompt
+        when={!isSaved && !saved}
+        message="Some Changes have not been saved. Are you sure you want to leave without saving them?"
+        />
+      {homeRedirect && <Redirect to="/" />}
+      {errorResponse ? (
+        <div className='tooltiptext-error response-error'>
+          <Trans i18nKey='errorResponse' className='response-error'>{{errorResponse}}</Trans>
+        </div>
+      ) : (
+        <div className='application-form-wrap'>
+          <TopBar
+            title={setNominate ? "Application Form" : "Review Form"}
+            t={t} />
+          <div
+            style={{ textAlign: 'end', width: '61%' }}
+            className='add-section-btn-wrapper'
+            >
+            <button
+              className='add-section-btn'
+              data-title="Add Section"
+              onMouseUp={() => addSection()}
+              >
+              <i className="fas fa-plus fa-lg add-section-icon"></i>
+            </button>
+            <input
+              type="button"
+              value="Save"
+              style={!isReview ? { marginTop: '16em'} : {}}
+              className='save-form-btn'
+              data-title="Save"
+              onClick={handleSave}
+              disabled={isSaveDisabled || disableSaveBtn}
+              />
+          </div>
+          <div className="application-form-wrapper">
+            {setNominate && (
```

Comment:
All this logic that depends on setNominate is specific to the application form, so I'd expect to find it in ApplicationForm.js and not here.